plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -39,6 +61,102 @@ export type EntityRefLinkProps = {
   children?: React.ReactNode;
 } & Omit<LinkProps, 'to'>;
 
+type PeekAheadPopoverProps = {
+  popupState: PopupState;
+  entityRef: CompoundEntityRef;
+};
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '20em',
+    },
+  };
+});
+
+export const PeekAheadPopover = ({
+  popupState,
+  entityRef,
+}: PeekAheadPopoverProps) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+  const [entity, setEntity] = useState<Entity | undefined>();
+
+  const { loading, error } = useAsync(async () => {
+    if (!entity) {
+      if (popupState.isOpen) {
+        const catalogApi = apiHolder.get(catalogApiRef);
+        if (catalogApi) {
+          const retrievedEntity = await catalogApi.getEntityByRef(entityRef);
+          if (!retrievedEntity) {
+            throw new Error(`${entityRef.name} was not found`);
+          }
+          setEntity(retrievedEntity);
+        }
+      }
+    }
+  }, [popupState.isOpen, apiHolder, entityRef]);
+
+  return (
+    <HoverPopover
+      PaperProps={{
+        className: classes.popoverPaper,
+      }}
+      {...bindPopover(popupState)}
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'center',
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+    >
+      <Card>
+        {loading && <Progress />}
+        <CardContent>
+          <Typography color="textSecondary">{entityRef.namespace}</Typography>
+          <Typography variant="h5" component="div">
+            {entityRef.name}
+          </Typography>
+          <Typography color="textSecondary">{entityRef.kind}</Typography>
+          <Typography variant="body2">
+            {error && <Alert severity="warning">{error.message}</Alert>}
+            {entity && (
+              <>
+                {entity.metadata.description}
+                <br />
```

Comment:
Did you know about the `paragraph` prop on `Typography`? That's what you should use to form properly paragraph formed things and get the right amount of spacing after.

And these should just be two separate Typography elements.

I see you also put the Alert inside, the Typography, you're mixing and matching a bit here and you'll end up with browser warnings that you can't put `<p>` or `<div>` inside `<span>` and stuff like that.

You should probably look more intentionally at margin usage, something like

```tsx
<Typography color="textSecondary" paragraph>{entityRef.kind}</Typography>
{error && <Alert .... />}
{entity && (
  <>
    <Typography paragraph>{description...
    <Typograph>{type...  // no paragraph here on the last one if it doesn't want to have more margin below
  </>
)}
```

and if you need good old classical margins between things, I recommend that you wrap them in a mui `<Box marginTop={2}>` or similar.